### PR TITLE
feat(platform): get file branchOrTag

### PR DESCRIPTION
**Pull request by olegkrivtsov**
## Changes:

Refactor `getJsonFile()`: add the optional branchOrTag, which is unused at platforms, unify null handling

## Context:

As suggested by @viceice in https://github.com/renovatebot/renovate/pull/12514#pullrequestreview-802204058

Related to https://github.com/renovatebot/renovate/issues/2496

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/platform/github/index.ts [hunk 1]
```diff
@@ -173,9 +173,11 @@ async function getBranchProtection(
 
 export async function getRawFile(
   fileName: string,
-  repo: string = config.repository
+  repo?: string,
+  branchOrTag?: string
 ): Promise<string | null> {
-  const url = `repos/${repo}/contents/${fileName}`;
+  const repoName = repo ?? config.repository;
```
Comment (viceice) [inline lib/platform/github/index.ts:179]: Please verify that all platforms do the same

File: lib/platform/azure/index.ts [hunk 2]
```diff
@@ -113,15 +113,16 @@ export async function getRepos(): Promise<string[]> {
 
 export async function getRawFile(
   fileName: string,
-  repoName?: string
+  repo?: string,
```
[the remark this answers is not in the document]
Comment (rarkins) [reply lib/platform/azure/index.ts:116]: Doesn't make too much difference to me, but we should be consistent. Looks like our types definition uses `repo`

[the remark this answers is not in the document]
Comment (viceice) [reply lib/platform/azure/index.ts:116]: sure, but we use `fileName`, so `repoName` would be more aligned i think.

[the remark this answers is not in the document]
Comment (rarkins) [reply lib/platform/azure/index.ts:116]: In that case we should do a separate refactor for consistency prior to this PR

**Review by viceice — APPROVED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**